buttercup/cogs/history.py

```diff
@@ -563,13 +565,109 @@ async def _rate(
             file=discord_file,
         )
 
+    async def _get_user_progress(
+        self, user: Dict[str, Any], start: datetime, time_frame: timedelta
+    ) -> int:
+        # We ask for submission completed by the user in the time frame
+        # The response will contain a count, so we just need 1 result
+        progress_response = self.blossom_api.get(
+            "submission/",
+            params={
+                "completed_by": user["id"],
+                "from": (start - time_frame).isoformat(),
+                "page_size": 1,
+            },
+        )
+        if progress_response.status_code != 200:
+            raise RuntimeError("Failed to get progress")
+        return progress_response.json()["count"]
+
+    async def _until_user(
+        self,
+        msg: SlashMessage,
+        user: Dict[str, Any],
+        target_username: str,
+        start: datetime,
+    ) -> None:
+        """Determine how long it will take the user to catch up with the target user."""
+        # Try to find the target user
+        target_response = self.blossom_api.get_user(target_username)
+        if target_response.status != BlossomStatus.ok:
+            raise InvalidArgumentException("goal", target_username)
+
+        target = target_response.data
+
+        if user["gamma"] > target["gamma"]:
+            # Swap user and target
+            temp = user

```

Comment:
You don't need a temp variable in Python:

```
x = 1
y = 2

x, y = y, x

print(x, y)
```

It'll return 2, 1.

Reply:
But also, do we need to swap them at all?

Reply:
Ohhh that's smart!

Yes we need to swap them, the target needs to be the user with the higher gamma, otherwise the prediction calculation and the output will be misleading.